### PR TITLE
Replace deprecated ioctl crate with nix

### PR DESCRIPTION
**Pull request by flosse**
@birkenfeld Could you have a look at the used nix macros? At the moment I can't test the code :-\

### CODE REVIEW

File: ethercat-sys/src/lib.rs [hunk 1]
```diff
@@ -7,101 +7,101 @@
...
+    ioctl_readwrite!( CONFIG              , EC, 0x19, ec_ioctl_config_t);
+    ioctl_readwrite!( CONFIG_PDO          , EC, 0x1a, ec_ioctl_config_pdo_t);
+    ioctl_readwrite!( CONFIG_PDO_ENTR     , EC, 0x1b, ec_ioctl_config_pdo_entry_t);
+    ioctl_readwrite!( CONFIG_SDO          , EC, 0x1c, ec_ioctl_config_sdo_t);
+    ioctl_readwrite!( CONFIG_IDN          , EC, 0x1d, ec_ioctl_config_idn_t);
+    ioctl_readwrite!( EOE_HANDLER         , EC, 0x1e, ec_ioctl_eoe_handler_t);
+    ioctl_none!     ( REQUEST             , EC, 0x1f);
+    ioctl_none!     ( CREATE_DOMAIN       , EC, 0x20);
+    ioctl_readwrite!( CREATE_SLAVE_CONFIG , EC, 0x21, ec_ioctl_config_t);
+    ioctl_write_ptr!( SELECT_REF_CLOCK    , EC, 0x22, u32);
```
Comment (birkenfeld) [inline ethercat-sys/src/lib.rs:50]: This one might need to be `ioctl_write_int`. (This ioctl isn't currently used in the Rust code.)

File: ethercat-sys/src/lib.rs [hunk 2]
```diff
@@ -7,101 +7,101 @@
...
+    ioctl_read!     ( ACTIVATE            , EC, 0x23, ec_ioctl_master_activate_t);
+    ioctl_none!     ( DEACTIVATE          , EC, 0x24);
+    ioctl_write_int!( SEND                , EC, 0x25);
```
Comment (birkenfeld) [inline ethercat-sys/src/lib.rs:53]: This looks incorrect; the C header uses `EC_IO()` which would translate to `ioctl_none` while `ioctl_write` uses `IOW`.

The case here seems to be "strange" and you need to use `ioctl_write_int_bad!(SEND, request_code_none!(EC, 0x25))`.

File: ethercat-sys/src/lib.rs [hunk 3]
```diff
@@ -7,101 +7,101 @@
...
+    ioctl_write_ptr!( SC_EMERG_SIZE       , EC, 0x3a, ec_ioctl_sc_emerg_t);
+    ioctl_readwrite!( SC_EMERG_POP        , EC, 0x3b, ec_ioctl_sc_emerg_t);
+    ioctl_write_ptr!( SC_EMERG_CLEAR      , EC, 0x3c, ec_ioctl_sc_emerg_t);
+    ioctl_readwrite!( SC_EMERG_OVERRUNS   , EC, 0x3d, ec_ioctl_sc_emerg_t);
+    ioctl_readwrite!( SC_SDO_REQUEST      , EC, 0x3e, ec_ioctl_sdo_request_t);
+    ioctl_readwrite!( SC_REG_REQUEST      , EC, 0x3f, ec_ioctl_reg_request_t);
+    ioctl_readwrite!( SC_VOE              , EC, 0x40, ec_ioctl_voe_t);
+    ioctl_readwrite!( SC_STATE            , EC, 0x41, ec_ioctl_sc_state_t);
+    ioctl_write_ptr!( SC_IDN              , EC, 0x42, ec_ioctl_sc_idn_t);
+    ioctl_write_int!( DOMAIN_SIZE         , EC, 0x43);
```
Comment (birkenfeld) [inline ethercat-sys/src/lib.rs:83]: same here

**Comment by birkenfeld** [comment]
The `ioctl-sys` crate doesn't seem to be actually deprecated. Only `ioctl` is (but they are unaffiliated, strangely).

**Comment by flosse** [comment]
> The `ioctl-sys` crate doesn't seem to be actually deprecated. Only `ioctl` is (but they are unaffiliated, strangely).

You're right, `ioctl-sys` is officially not deprecated but the latest release was  Mar 17, 2017 and the latest commit was 20 Apr 2017 whereas the latest nix was released Feb 4, 2020 and is maintained by a large number of contributors.

**Comment by birkenfeld** [comment]
Sure, but it's small and stable, there's no outstanding issues in the repo, and I don't believe that you have to put out releases just to show activity. `nix`'s scope is much, much broader so there's no surprise it needs (and gets) much more maintenance.

Still, maybe we'll need `nix` for other things in the future, so I'm fine with this (after comments are addressed).  (Can't test with real HW at the moment either, though.)

**Review by birkenfeld — CHANGES_REQUESTED** [review]

**Comment by flosse** [comment]
> Sure, but it's small and stable, there's no outstanding issues in the repo

You're right. So lets ignore my PR ;-)
If there will be any problems in the future we can discuss again.

**Pull request closed without merging**